core/server/server.js

```diff
@@ -421,6 +422,11 @@ class Server {
       cert,
       key,
     }))
+    camp.handle(function (req, res, next) {
+      const frontendMaxAge = 300
+      setHeadersForCacheLength(res, frontendMaxAge)
+      next()
```

Comment:
We only register the handler once at server startup - we're not re-registering it on every request
but the actual handler function _does_ run on every request. i.e: we are re-evaluating `new Date()` for the `Expires`/`Date` headers on every request - we're not just evaluating `new Date()` once on server startup and using that value for the server lifetime.

Reply:
I guess that means these values get set on the requests first, and then it's up to the badge request handler to overwrite them when returning badges.

That seems okay for the most part, though for a few kinds of requests that might be a problem: static badges, redirects, and the suggest endpoint. Static badges and redirects invoke `setCacheHeadersForStaticResource()` which sets `Cache-Control` and `Last-Modified` and after this change they get a hodgepodge of headers. I haven't dug into the suggest endpoint, though I would guess those requests are not getting any cache headers in master. Probably it would be a good idea to add server tests which check all the headers which are set on those requests.

It seems better to set the new cache headers only on requests being fulfilled through the scoutcamp documentRoot (i.e. [`staticUnit`](https://github.com/espadrine/sc/blob/c1042e58fb849e1a2ec87dc22695d21329b0e8e1/lib/camp.js#L698-L703)), instead of on every request. Looking at the code, I don't see an easy way to do that without modifying scoutcamp.

In terms of the cache behavior itself, an alternative to the five-minute cache would be to use `setCacheHeadersForStaticResource()` (maybe with a shorter timeout) combined with `serverHasBeenUpSinceResourceCached()`, which most of the time will send 304s. That'd be closer aligned to the way we handle our existing static content.

Reply:
> I guess that means these values get set on the requests first, and then it's up to the badge request handler to overwrite them when returning badges.

Yeah that's what I was trying to explain when I said

> What this actually does is apply `frontendMaxAge` to everything, but then for every other request we then overwrite it with subsequent calls to `setHeader()`

clearly that failed.

Its useful you point out that there is an issue with static badges and redirects. I'll revisit. It has also occurred to me that we might want this on in production, but not for development.

> It seems better to set the new cache headers only on requests being fulfilled through the scoutcamp documentRoot

Yes that would be ideal, but not available out-of-the-box. Maybe we _do_ need a PR to scoutcamp after all. I will rethink it.

Reply:
> Yeah that's what I was trying to explain when I said
> 
> > What this actually does is apply `frontendMaxAge` to everything, but then for every other request we then overwrite it with subsequent calls to `setHeader()`

Ahha sorry, I missed that in the top comment.